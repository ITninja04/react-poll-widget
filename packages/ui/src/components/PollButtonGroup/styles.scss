@import "../../colors";

.poll-button-group {
  button.ui.button {
    width: 0;
  }

  button.ui.button:not(:last-child) {
    border-right: 1px solid $darkwhite;
  }
}
