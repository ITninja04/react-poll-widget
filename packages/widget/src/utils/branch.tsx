export type BranchFunction = (condition: boolean, ComponentIfTrue: React.ReactNode, ComponentIfFalse: React.ReactNode) => React.ReactNode;

export const branch: BranchFunction = (condition, ComponentIfTrue, ComponentIfFalse) => condition
  ? ComponentIfTrue
  : ComponentIfFalse;