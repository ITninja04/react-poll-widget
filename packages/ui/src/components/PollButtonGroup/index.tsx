import React from 'react';
import { Button } from 'semantic-ui-react';

import './styles.scss';

type PollButton = {
    content: string;
    onClick: () => void
}

const PollButtonGroup: React.FC<{
    buttons: PollButton[],
    icons: boolean
}> = ({
    buttons,
    icons
}) => (
            <Button.Group 
            fluid 
            icon={icons}
            className='poll-button-group'
            >
                {
                    buttons.map(button => (
                        <Button onClick={button.onClick}>
                            {button.content}
                        </Button>
                    ))
                }
            </Button.Group>
        )

export default PollButtonGroup