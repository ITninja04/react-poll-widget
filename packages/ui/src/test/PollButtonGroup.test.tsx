import PollButtonGroup from "../components/PollButtonGroup";
import { createSnapshotTest } from "./utils";

createSnapshotTest(PollButtonGroup, {
  options: [
    { id: '1', value: 'Yes', label: 'Yes' },
    { id: '2', value: 'No', label: 'No' }
  ]
}, 'Yes/No')

createSnapshotTest(PollButtonGroup, {
  options: [
    { id: '1', value: 'Yes', label: 'Yes' },
    { id: '2', value: 'No', label: 'No' }
  ],
  loading: true
}, 'Loading')