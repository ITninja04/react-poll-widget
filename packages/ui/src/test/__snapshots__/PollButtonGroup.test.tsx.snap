// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Snapshot) Loading should render 1`] = `
<DocumentFragment>
  <div
    class="ui fluid buttons poll-button-group"
  >
    <button
      class="ui button"
    >
      Yes
    </button>
    <button
      class="ui button"
    >
      No
    </button>
  </div>
</DocumentFragment>
`;

exports[`(Snapshot) Yes/No should render 1`] = `
<DocumentFragment>
  <div
    class="ui fluid buttons poll-button-group"
  >
    <button
      class="ui button"
    >
      Yes
    </button>
    <button
      class="ui button"
    >
      No
    </button>
  </div>
</DocumentFragment>
`;
